public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt

```diff
@@ -171,16 +151,31 @@ data class CustomerInfo constructor(
                 "nonSubscriptionTransactions: $nonSubscriptionTransactions,\n" +
                 "requestDate: $requestDate\n>"
 
-    override fun hashCode(): Int {
-        var result = entitlements.hashCode()
-        result = 31 * result + nonSubscriptionTransactions.hashCode()
-        result = 31 * result + allExpirationDatesByProduct.hashCode()
-        result = 31 * result + allPurchaseDatesByProduct.hashCode()
-        result = 31 * result + requestDate.hashCode()
-        result = 31 * result + jsonObject.hashCode()
-        result = 31 * result + schemaVersion
-        result = 31 * result + firstSeen.hashCode()
-        result = 31 * result + originalAppUserId.hashCode()
-        return result
-    }
+    override fun equals(other: Any?) = other is CustomerInfo && ComparableData(this) == ComparableData(other)
+    override fun hashCode() = ComparableData(this).hashCode()
+}
+
+/**
+ * Contains fields to be used for equality, which ignores requestDate and jsonObject.
```

Comment:
Perhaps we can add a note specifying the intent of ignoring `requestDate` and `jsonObject`? 
i.e.: `requestDate` is excluded so that two `customerInfo` objects that are otherwise identical won't be considered different if they were refreshed at a different point in time, and `jsonObject` is excluded because we're already using the parsed fields for comparisons

Reply:
Good idea 👍